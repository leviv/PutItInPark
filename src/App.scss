/* =================
  Variables
  ================= */
$primary-color: #2A9D8F;
$secondary-color: #FED892;
$third-color: #264653;

/* =================
  General Styles
  ================= */
* {
  font-family: 'Vollkorn', serif;
}

h1, h2, h3, h4, h5, h6 {
  color: $third-color;
  font-family: 'Unica One', serif;
  text-transform: uppercase;
}

h1 {
  margin: 20px 0;
  span {
    background-color: $third-color;
    color: $secondary-color;
    font-family: 'Unica One', serif;
    padding: 3px 8px;
  }
}

button.button {
  border-radius: 20px;
  bottom: 0px;
  font-family: 'Unica One', serif;
  font-size: 20px;
  height: 40px;
  margin: 3px 10px;
  padding: 2px 20px;
  position: relative;
  text-transform: uppercase;
  transition: all .2s;
  -webkit-transition: all .2s;
  background: #fff;

  &.button-primary {
    border: 3px solid $primary-color;
    color: $third-color;

    &:hover {
      background: $primary-color;
      color: $secondary-color;
    }
  }

  &.button-secondary {
    border: 3px solid $secondary-color;
    color: $third-color;

    &:hover {
      background: $secondary-color;
    }
  }

  &:hover {
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    bottom: 2px;
  }
}

body, html, #root {
  height: 100%;
}

body {
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizelegibility;
  font-family: "Dosis", sans-serif;
  padding-top: 56px
}

h2 {
  margin: 20px 0;
}

footer {
  background-color: $third-color;
  padding: 15px;
  h3 {
    color: $secondary-color;
  }
}

/* =================
  Splash Page styles
   ================= */

#splash .row {
  align-items: center;
  background: linear-gradient(60deg, $third-color 50%, #fff 50%);
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;

  .splash-image {
    height: 100%;

    img {
      width: 100%;
    }
  }

  .splash-title {
    height: 100%;
    margin: 30px;
    padding: 15px;
  }
}

#models-splash {
  .model-splash {
    margin: 0 0 20px 0;

    img {
      width: 100%;
      margin: 10px 0;
    }
  }
}

#splash-call-to {
  background: $secondary-color;
  text-align: center;

  .splash-call-to-content {
    padding: 25px 0;
  }
}

/* =================
  About
  ================= */

.about-lead {
  margin: 25px 0;
  text-align: center;
  font-size: 20px;
  background: #eee;
  padding: 15px;
}

.team-member {
  background: $third-color;
  border-radius: 8px;
  bottom: 0;
  color: #fff;
  margin: 10px;
  position: relative;
  transition: all .2s;
  -webkit-transition: all .2s;

  h3 {
    color: $secondary-color;
    word-wrap: break-word;
    word-spacing: 100vw;
  }

  img {
    width: 100%;
  }

  .team-desc {
    padding: 10px;
  }

  &:hover {
    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
    bottom: 3px;
  }
}

/* =================
  Model instance page styles
  ================= */

.instance-intro {
  background-color: $primary-color;
  background-position: center center;
  background-attachment: fixed;
  background-size: cover;
  padding: 100px 0;
  margin-bottom: 20px;

  h1 {
    text-align: center;
    span {
      background-color: $primary-color;
    }
  }
}

.instance {
  .state p, .dates-open p, .reservable p, .stay-limit p, .activities-list p {
    text-transform: capitalize;
  }

  .map-container {
    height: 300px;
    margin-bottom: 50px;
    position: relative;
    width: 100%;
  }
}

/* =================
  Grid Card Styles
  ================= */

.model-intro {
  background-color: $secondary-color;
  text-align: center;

  h2 {
    padding: 60px 0;
    text-decoration: underline;
    margin: 0;
  }
}

.instance-container {
  width: 100%;

  .instance-overlay {
    background: rgb(255,255,255);
    background: linear-gradient(180deg, rgba(255,255,255,0) 0%, rgba(255,255,255,0) 30%, rgba($third-color,.9) 50%, rgba($third-color, .9) 100%);
    background-size: 200% 200%;
    border-radius: 5px;
    height: 100%;
    width: 100%;
    transition: all .2s;
    -webkit-transition: all .2s;

    .hidden-details {
      color: #fff;
      padding: 15px;
      text-transform: capitalize;
      visibility: hidden;
    }

    &:hover {
      background-position: 50% 100%;

      .hidden-details {
        visibility: visible;
      }
    }
  }

  .instance-card {
    background-color: $primary-color;
    background-position: center center;
    background-size: cover;
    border-radius: 5px;
    bottom: 0;
    display: inline-block;
    height: 250px;
    margin: 20px 0;
    position: relative;
    transition: all .2s;
    -webkit-transition: all .2s;
    width: 100%;

    &.small-card {
      height: 150px;
    }

    .details {
      bottom: 10px;
      color: #fff;
      position: absolute;
      text-transform: capitalize;
      width: 100%;

      h4 {
        text-align: center;
        color: $secondary-color;
      }
    }

    &:hover {
      box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
      bottom: 3px;
    }
  }
}

/* =================
  Pagination
  ================= */

ul.pagination {
  display: inline-block;
  padding-left: 15px;
  padding-right: 15px;
  text-align: center;
}

ul.pagination li {
  display: inline-block;
}

ul.pagination a {
  border: 1px solid $third-color;
  color: $third-color;
  cursor: pointer;
  display: inline-block;
  font-size: 18px;
  margin: 0 5px;
  padding: 3px;
  transition: all .1s;

  &:hover {
    background-color: $third-color;
    color: $secondary-color;
  }
}

/* =================
  Sorting, searching and filtering bar
  ================= */

#augment-container {
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  border-radius: 5px;
  -webkit-border-radius: 5px;
  height: 98px;
  margin: 30px 0;
  overflow: hidden;
  padding: 15px 35px;
  transition: all .2s;
  background-color: $third-color;

  .search-row {
    align-items: center;
    display: flex;
    position: relative;

    #carat {
      border-radius: 50%;
      -webkit-border-radius: 50%;
      color: $secondary-color;
      cursor: pointer;
      font-size: 30px;
      height: 50px;
      padding: 10px;
      position: absolute;
      right: 30px;
      width: 50px;
      transition: all .2s;

      &:hover {
        background: $primary-color;
      }
    }
  }

  .model-search {
    align-items: center;
    background: $primary-color;
    border-radius: 25px;
    color: $secondary-color;
    display: flex;
    padding: 5px 15px;
    margin: 15px 0 15px 15px;
    width: fit-content;

    .model-search-component {
      border: none;
      display: inline;
    }

    input {
      background: $primary-color;
      color: #fff;
      max-width: 300px;

      &::placeholder {
        color: #ddd;
      }
    }

    h4 {
      color: $secondary-color;
      margin-bottom: 0;
    }

    a {
      color: $secondary-color;
    }
  }

  h4 {
    color: $secondary-color;
  }

  button {
    background-color: $third-color;
    color: $secondary-color
  }

  select {
    background: $third-color;
    color: $secondary-color;
    display: inline;
    margin: 8px 10px 8px 0;
    max-width: 120px;
    padding: 50px;
  }
}
