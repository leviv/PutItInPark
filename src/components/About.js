import React from 'react';
import TeamMember from './TeamMember'
import BillyPhoto from '../assets/img/Billy.png'
import EllaPhoto from '../assets/img/Ella.png'
import JosephPhoto from '../assets/img/Joseph.png'
import LeviPhoto from '../assets/img/Levi.png'
import MilesPhoto from '../assets/img/Miles.png'

// Gitlab issues/commit retrieval inspired by shub95/foodmeonce
class About extends React.Component {
  constructor(props) {
    super(props)
    var Billy = [0,0,0]
    var Ella = [0,0,0]
    var Joseph = [0,0,0]
    var Levi = [0,0,14]
    var Miles = [0,0,0]
    var totalCommits = 0
    var closedIssues = 0

    this.state = {
      Billy: Billy,
      Ella: Ella,
      Joseph: Joseph,
      Levi: Levi,
      Miles: Miles,
      totalCommits: totalCommits,
      closedIssues: closedIssues
    }
  }

  componentDidMount() {
    this.grabCommits();
    this.grabIssues();
  }

  async grabIssues(){
    fetch("https://gitlab.com/api/v4/projects/14563233/issues?per_page=1000&page=1")
      .then(res => res.json())
      .then(res => {
        res.forEach(issue => {
          this.state.totalIssues +=1
          if(issue.closed_by != null){
            this.state.closedIssues +=1
            issue.assignees.forEach(assignee => {
              switch (assignee.name){
                case "Ella Robertson":
                case "ellarobertson":
                  this.state.Ella[1] +=1
                  break;
                case "Levi Villarreal":
                  this.state.Levi[1]+=1
                  break;
                case "Joseph Engelhart":
                case "jengelhart":
                    this.state.Joseph[1]+=1
                    break;
                case "Miles Chandler":
                    this.state.Miles[1]+=1
                    break;
                case "Billy Vo":
                case "billyvo":
                  this.state.Billy[1]+=1
                  break;
                default:
                  console.log(issue)
              }
            })
          }
        });
        this.setState(this.state)
      })
  }

  async grabCommits(){
    var page = 0
    while(page < 4){
    page+=1
    fetch("https://gitlab.com/api/v4/projects/14563233/repository/commits?per_page=100&page=" + page)
        .then(res => res.json())
        .then(res => {

          res.forEach(commit => {
            this.state.totalCommits += 1
            switch (commit.committer_name){
              case "Ella Robertson":
              case "ellarobertson":
                this.state.Ella[0] +=1
                break;
              case "Levi Villarreal":
                this.state.Levi[0]+=1
                break;
              case "Joseph Engelhart":
              case "jengelhart":
                  this.state.Joseph[0]+=1
                  break;
              case "Miles Chandler":
                  this.state.Miles[0]+=1
                  break;
              case "Billy Vo":
              case "billyvo":
                this.state.Billy[0]+=1
                break;
              default:
            }
          });
        })
      }
      this.setState(this.state)
  }

  render() {
    const {
      Billy,
      Ella,
      Joseph,
      Levi,
      Miles,
      totalCommits,
      closedIssues
    } = this.state

    return (
      <React.Fragment>
        <div className="model-intro">
          <h2>About</h2>
        </div>

        <div className="container">

          <p class="about-lead">Purpose: This website is meant to encourage Americans/tourists to go outdoors by highlighting National Parks and recreational activities in their area. Real beauty is found outside and we think more people should spend their time enjoying it.</p>

          <div className="about-buttons text-center">
            <a href="https://gitlab.com/leviv1/putitinpark/" target="_blank" rel="noopener noreferrer"><button className="button button-primary">Gitlab Repo</button></a>
            <a href="https://documenter.getpostman.com/view/9011044/SVtR19mz?version=latest" target="_blank" rel="noopener noreferrer"><button className="button button-secondary">API Docs</button></a>
          </div>

          <br/>

          <h1 className="text-center"><span>Total Stats</span></h1>

          <h3 className="text-center">Total Commits: {totalCommits}</h3>
          <h3 className="text-center">Closed issues: {closedIssues}</h3>

          <div className="row">
            <TeamMember
              name="Miles Chandler"
              imageUrl={MilesPhoto}
              commits={Miles[0]}
              issuesClosed={Miles[1]}
              tests={Miles[2]}
            />
            <TeamMember
              name="Joseph Engelhart"
              imageUrl={JosephPhoto}
              commits={Joseph[0]}
              issuesClosed={Joseph[1]}
              tests={Joseph[2]}
            />
            <TeamMember
              name="Ella Robertson"
              imageUrl={EllaPhoto}
              commits={Ella[0]}
              issuesClosed={Ella[1]}
              tests={Ella[2]}
            />
          </div>
          <div className="row">
            <div className="col-sm-2"></div>
            <TeamMember
              name="Levi Villarreal"
              imageUrl={LeviPhoto}
              commits={Levi[0]}
              issuesClosed={Levi[1]}
              tests={Levi[2]}
            />
            <TeamMember
              name="Billy Vo"
              imageUrl={BillyPhoto}
              commits={Billy[0]}
              issuesClosed={Billy[1]}
              tests={Billy[2]}
            />
            <div className="col-sm-2"></div>
          </div>
          <p className="about-lead">About this site: This site is supported by Google Cloud serving a Flask backend and React frontend.</p>
        </div>
      </React.Fragment>
    );
  }
}

export default About;
