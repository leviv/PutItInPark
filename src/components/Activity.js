import React from 'react';
import { Link, Route } from 'react-router-dom';
import NotFound from './NotFound';

const activities = {
  "climbing" : {
    name: "climbing",
    imageUrl: "../assets/img/climbing.jpg",
    fees: "$100 per day",
    datesOpen: "year round",
    locations: ["yellowstone", "yosemite"],
  },
  "rafting" : {
    name: "climbing",
    imageUrl: "../assets/img/climbing.jpg",
    fees: "$150 per day",
    datesOpen: "year round",
    locations: ["yellowstone", "yosemite"],
  },
  "camping" : {
    name: "camping",
    imageUrl: "../assets/img/camping.jpg",
    fees: "$250 per day",
    datesOpen: "year round",
    locations: ["yosemite", "grand-canyon"],
  },
}

class Activity extends React.Component {
  render() {
    const { match } = this.props;
    const activity = activities[match.params.activityName]

    // Valid park
    if (activity !== undefined){
      return (
        <div className="container instance">
          <h2>{activity.name}</h2>

          <div className="picture">
            <img src={activity.imageUrl} alt="Cliff at Yosemite park"/>
          </div>

          <h3>Fees</h3>
          <p>{activity.fees}</p>

          <h3>Dates Open</h3>
          <p>{activity.datesOpen}</p>

          <h3>Parks with these activities</h3>
          <ul>
            {activity.locations.map(function(location){
              return (
                <li><p>{location}</p></li>
              );
            })}
          </ul>
        </div>
      );
    }

    // invalid park name
    return (
      <Route component={NotFound} />
    );
  }
}

export default Activity;
