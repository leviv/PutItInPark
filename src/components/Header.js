import React from 'react';
import { Link } from "react-router-dom";
import Logo from '../assets/img/logo.png';

class Header extends React.Component {

  render() {
    return (
      <nav className="navbar fixed-top navbar-expand-lg navbar-light bg-light">
        <div className="container">
          <a className="navbar-brand" href="/"><img src={Logo} alt="Icon with mountains and the sun" width="40px"/></a>
          <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="navbarNav">
            <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                <Link className="nav-link about-link" to="/about/"><h5>About</h5></Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link parks-link" to="/parks/"><h5>National Parks</h5></Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link recs-link" to="/recreation/"><h5>Recreation</h5></Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link states-link" to="/states/"><h5>States</h5></Link>
              </li>
            </ul>

            <form className="form-inline my-2 my-lg-0">
              <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search"/>
              <button className="button button-primary my-2 my-sm-0" type="submit">Search</button>
            </form>
          </div>
        </div>
      </nav>
    );
  }
}

export default Header;
