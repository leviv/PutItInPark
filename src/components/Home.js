import React from 'react';
import { Link } from 'react-router-dom';
import Splash from '../assets/img/splash.png';
import Yosemite from '../assets/img/yosemite.jpg';
import Climbing from '../assets/img/climbing.jpg';
import Texas from '../assets/img/texas.jpg';

class Home extends React.Component {
  render() {
    return (
      <React.Fragment>
        <section id="splash">
          <div className="row">
            <div className="col-md-7 splash-image">
              <img src={Splash} alt="Illustration of a man with binoculars"/>
            </div>
            <div className="col-md-5">
              <div className="splash-title">
                <h1><span>Put It In Park</span></h1>
                <p>Explore different national parks around the country, and everything that they have to offer!</p>
                <Link to="/about/"><button className="button button-primary">About</button></Link>
                <Link to="/parks/"><button className="button button-primary">Parks</button></Link>
              </div>
            </div>
          </div>
        </section>

        <section id="models-splash">
          <div className="container">
            <h1 className="text-center">Explore Parks, States, and Activities!</h1>
            <div className="row">
              <div className="col-md-4 model-splash">
                <img src={Yosemite} alt="National park landscape"/>
                <h3>National Parks</h3>
                <p>Look through all of the national parks that the United States has to offer! View information about each park such as what recreational activities the park has, some basic history, and the location so that you can one day visit yourself.</p>
                <Link to="/parks/"><button className="button button-secondary">Explore Parks</button></Link>
              </div>
              <div className="col-md-4 model-splash">
                <img src={Climbing} alt="A man climbing outdoors"/>
                <h3>Recreational Activities</h3>
                <p>From climbing to whitewater rafting, there are a host of recreational activities that one can experience at national parks. Find your favorite activity, and view where in the world you can go to experience it for yourself.</p>
                <Link to="/recreation/"><button className="button button-secondary">Explore Activities</button></Link>
              </div>
              <div className="col-md-4 model-splash">
                <img src={Texas} alt="Texas state flag"/>
                <h3>States</h3>
                <p>View information on each of the 50 United States such as thier population, the recreational activities that they have, as well as the different national parks that each state has to offer.</p>
                <Link to="/states/"><button className="button button-secondary">Explore States</button></Link>
              </div>
            </div>
          </div>
        </section>

        <section id="splash-call-to">
          <div className="splash-call-to-content">
            <h2>Find your favorite national park today!</h2>
          </div>
        </section>
      </React.Fragment>
    );
  }
}

export default Home;
