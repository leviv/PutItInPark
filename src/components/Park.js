import React from 'react';
import { Link, Route } from 'react-router-dom';
import { Map, Marker, GoogleApiWrapper } from 'google-maps-react';
import ActivityCard from './ActivityCard';
import NotFound from './NotFound';

const API_ENDPOINT = "https://flask-backend-dot-potent-retina-254722.appspot.com";

class Park extends React.Component {
  constructor(props) {
    super(props);
    const { match } = this.props;
    const parkName = match.params.parkName

    this.state = {
      park: [],
      recreations: [],
      parkName: parkName,
      parkLoaded: false,
      loaded: false
    };
  }

  makeApiCall(parkName) {
    fetch(API_ENDPOINT + "/nationalparks/" + parkName)
      // Transform the data into json
      .then((resp) => resp.json())
      .then((data) => {
        // Process data
        data.rec_ids = data.rec_ids.split(",");
        this.state.park = data;
      }).then(() => {
        let numRecsLoaded = 0;

        // Check if we need to load any recreations
        if (this.state.park.rec_ids.length === 0) {
          this.setState({loaded: true});
        }

        // Get the recreational information
        this.state.park.rec_ids.forEach((id) => {
          fetch(API_ENDPOINT + "/recreations/id=" + id)
            // Transform the data into json
            .then((resp) => resp.json())
            .then((data) => {
              // Process data
              this.state.recreations.push(data);
            }).then(() => {
              numRecsLoaded++;
              if (numRecsLoaded === this.state.park.rec_ids.length) {
                this.setState({loaded: true});
              }
            });
        });
      });
  }

  componentDidMount() {
    this.makeApiCall(this.state.parkName);
  }

  render() {
    // The API call has finished
    if (this.state.loaded){
      const displayName = this.state.park.park_name.replace(/\-+/g, ' ');

      const row = this.state.recreations.map((x,i) => {
        return i % 4 === 0 ? this.state.recreations.slice(i, i+4) : null;
      }).filter(x => x != null);

      return (
        <React.Fragment>
          <div className="instance-intro"
               style={{ backgroundImage: `url(${this.state.park.imglink})`}}>
            <h1><span>{displayName}</span></h1>
          </div>

          <div className="container instance">
            <h3>Description</h3>
            <p>{this.state.park.description}</p>
            <p>{this.state.park.weather}</p>

            <div className="row">
              <div className="col-md-4 state">
                <h3>State</h3>
                <p><Link to={getSlug('state', this.state.park.location)}>{this.state.park.location}</Link></p>
              </div>
              <div className="col-md-4 fees">
                <h3>Park Fees</h3>
                <p>${this.state.park.fee} admission</p>
              </div>
              <div className="col-md-4 visitors">
                <h3>Annual Visitors</h3>
                <p>{this.state.park.visitors}</p>
              </div>
            </div>

            <h3>Recreational Areas</h3>
            {row.map((result, index) => {
              return (
                <div className="row" key={index}>
                  {result.map((item, innerIndex) => {
                    return (
                      <div className="col-md-3 instance-container" key={innerIndex}>
                        <ActivityCard
                          name={item.rec_name}
                          imageUrl={item.imglink}
                          num_activities={item.num_activities}
                          reservable={item.reservable}
                          stay_limit={item.stay_limit}
                        />
                      </div>
                    );
                  })}
                </div>
              );
           })}

           <h3>Addresss</h3>
           <div className="map-container">
             <Map
               google={this.props.google}
               zoom={8}
               style={{width: '100%', height: '100%', position: 'relative'}}
               initialCenter={{ lat: this.state.park.lat, lng: this.state.park.lon}}
             >
               <Marker position={{ lat: this.state.park.lat, lng: this.state.park.lon}} />
             </Map>
           </div>
          </div>
        </React.Fragment>
      );
    }

    // invalid park name
    return (
      <Route component={NotFound} />
    );
  }
}

function getSlug(pre, str) {
  return "/" + pre + "/" + str.replace(/\s+/g, '-').toLowerCase();
}

export default GoogleApiWrapper({
 apiKey: ('AIzaSyD4KTXfspSV4uzzkjwDEzzWBfQguQ9tyqA')
})(Park);
