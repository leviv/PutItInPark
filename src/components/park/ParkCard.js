import React from 'react';
import { Link } from 'react-router-dom';
import { displayName, slugName } from '../helpers/Helpers.js';

class ParkCard extends React.Component {
  render() {
    const slug = slugName('/park/', this.props.park_name);
    const title = displayName(this.props.park_name);

    return (
      <Link to={slug}>
        <div className="instance-card"
             style={{ backgroundImage: `url(${this.props.imglink})`}}
             >
          <div className="instance-overlay">
            <div className="hidden-details">
              <p>State: {this.props.location}</p>
              <p>No. Rec. Areas: {this.props.num_rec}</p>
              <p>No. Visitors: {this.props.visitors}</p>
              <p>Fees: ${this.props.fee}</p>
            </div>
            <div className="details">
              <h4>{title}</h4>
            </div>
          </div>
        </div>
      </Link>
    );
  }
}

export default ParkCard;
