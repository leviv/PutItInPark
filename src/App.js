import React from 'react';
import './App.scss';
import { BrowserRouter as Router, Switch, Route, Redirect} from "react-router-dom";

// Import components
import Home from './components/Home';
import Header from './components/Header';
import Footer from './components/Footer';
import About from './components/About';
import Parks from './components/Parks';
import Recreations from './components/Recreations';
import States from './components/States';
import NotFound from './components/NotFound';
import Park from './components/Park';
import Recreation from './components/Recreation';
import State from './components/State';
import Search from './components/Search';

function App() {
  return (
    <Router>

      <Header></Header>

      <Switch>
        <Route path="/" exact component={Home} />
        <Route path="/about/" component={About} />
        <Route path="/parks/:page" component={Parks} />
        <Route path="/recreations/:page" component={Recreations} />
        <Route path="/states/:page" component={States} />
        <Redirect from='/parks/' to='/parks/1' />
        <Redirect from='/recreations/' to='/recreations/1' />
        <Redirect from='/states/' to='/states/1' />
        <Route path="/park/:parkName" component={Park} />
        <Route path="/recreation/:recName" component={Recreation} />
        <Route path="/state/:stateName" component={State} />
        <Route path="/search/:searchString" component={Search} />

        <Route path="*" component={NotFound} />
      </Switch>

      <Footer></Footer>
    </Router>
  );
}

export default App;
