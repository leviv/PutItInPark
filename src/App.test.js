import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import App from './App';

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() });

// Custom components
import About from './components/About';
import Activity from './components/Activity';
import Header from './components/Header';
import Home from './components/Home';
import Park from './components/Park';
import ParkCard from './components/ParkCard';
import Parks from './components/Parks';
import Recreation from './components/Recreation';
import State from './components/State';
import StateCard from './components/StateCard';
import States from './components/States';
import TeamMember from './components/TeamMember';

describe('App', function() {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders 1 navbar', () => {
    const app = shallow(<App />);
    expect(app.find('Header').length).toEqual(1);
  });
});

describe('About page', function() {
  const about = shallow(<About />);

  it('renders successfully', () => {
    expect(about).toMatchSnapshot();
  });

  it('renders 5 team cards', () => {
    expect(about.find('TeamMember').length).toEqual(5);
  });
});

describe('Splash page', function() {
  it('renders successfully', () => {
    const splash = shallow(<Home />);
    expect(splash).toMatchSnapshot();
  });
});

describe('Parks page', function() {
  const parks = shallow(<Parks />);

  it('renders successfully', () => {
    expect(parks).toMatchSnapshot();
  });

  it('cointains 9 ParkCards', () => {
    expect(parks.find('ParkCard').length).toEqual(9);
  });
});

describe('Park instance page', function() {
  const match = {params: {parkName: "Yosemite"}};

  it('renders successfully', () => {
    const park = shallow(<Park match={match}/>);
    expect(park.props().parkName).toMatchSnapshot();
  });
});

describe('States page', function() {
  const states = shallow(<States />);

  it('renders successfully', () => {
    expect(states).toMatchSnapshot();
  });

  it('cointains 9 StateCards', () => {
    expect(states.find('StateCard').length).toEqual(9);
  });
});

describe('State instance page', function() {
  const match = {params: {stateName: "Texas"}};

  it('renders successfully', () => {
    const state = shallow(<State match={match}/>);
    expect(state).toMatchSnapshot();
  });
});

describe('Recreation page', function() {
  const rec = shallow(<Recreation />);

  it('renders successfully', () => {
    expect(rec).toMatchSnapshot();
  });
});

describe('Activity instance page', function() {
  const match = {params: {activityName: "Climbing"}};

  it('renders successfully', () => {
    const activity = shallow(<Activity match={match}/>);
    expect(activity).toMatchSnapshot();
  });
});
