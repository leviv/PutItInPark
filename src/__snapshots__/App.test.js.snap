// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic functionality Tests About page renders successfully 1`] = `ShallowWrapper {}`;

exports[`Basic functionality Tests Splash page renders successfully 1`] = `ShallowWrapper {}`;

exports[`Instance Page Tests Park instance page renders successfully 1`] = `undefined`;

exports[`Instance Page Tests Recreation instance page renders successfully 1`] = `ShallowWrapper {}`;

exports[`Instance Page Tests State instance page renders successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests Parks page renders ParkCard successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests Parks page renders page successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests Recreation page renders RecreationCard successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests Recreation page renders successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests States page renders StateCard successfully 1`] = `ShallowWrapper {}`;

exports[`Model Page Tests States page renders successfully 1`] = `ShallowWrapper {}`;
